aten/src/ATen/native/Embedding.cpp

```diff
@@ -15,18 +15,18 @@ Tensor embedding(const Tensor & weight, const Tensor & indices,
                  int64_t padding_idx, bool scale_grad_by_freq, bool sparse) {
   TORCH_CHECK(weight.dim() >= 1, "'weight' must be at least 1-D");
   auto indices_arg = TensorArg(indices, "indices", 1);
-  checkScalarType("embedding", indices_arg, kLong);
+  checkScalarTypes("embedding", indices_arg, {kLong, kInt});
 
   // TODO: use tensor.index() after improving perf
   if (indices.dim() == 1) {
-    return weight.index_select(0, indices);
+    return weight.index_select(0, indices.to(kLong));
```

Comment:
haha, maybe we should make index_select to be int32 too, but that can wait till another day :)